microsoft/electionguard-api-python
# 🛡. Add authentication and authorization

[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue
*Link your PR to an issue*

Fixes #174 
Fixes #173 

### Description
Add authentication and authorization. This PR adds auth support but stubs out the actual auth implementation. Supports JWT bearer tokens and form based auth, along with a few administrative functions to add users and change passwords. It also includes examples on the user api's demonstrating how to authorize endpoints based on scopes but it does not add that functionality to all of the endpoints in the api.

### Testing
1. call `auth/login` with the default credentials
2. verify you can call `/user/me`
3. create a new user as an admin with the `voter` or some other scope
4. login as the new user and try to create a user, expect that it fails (since you aren't an admin)
